### PR TITLE
cumulated updates and pddl constant support

### PR DESCRIPTION
**Pull request by vacancy**

### CODE REVIEW

File: lisdf/parsing/pddl_j.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def transform(
 
 class PDDLVisitor(Transformer):
     def __init__(
-        self, sdf: C.LISDF, domain: C.PDDLDomain, problem: C.PDDLProblem
```
Comment (williamshen-nz) [inline lisdf/parsing/pddl_j.py:50]: nit: is `lisdf` a better variable name? I can't remember the details of the lisdf representation...

File: lisdf/parsing/pddl_j.py [hunk 2]
```diff
@@ -101,8 +101,8 @@ def type_definition(self, args):
                 self.domain.types[arg_name] = C.PDDLType(arg_name, None)
 
     @inline_args
-    def constants_definition(self, *args):
-        raise NotImplementedError()
```
Comment (williamshen-nz) [inline lisdf/parsing/pddl_j.py:105]: nit: type hint

File: lisdf/parsing/pddl_j.py [hunk 3]
```diff
@@ -144,30 +148,34 @@ def typedvariable(self, name, typename):
...
+                    model_name, lname, self.domain.types["sdf::link"]
+                )
+            elif name in self.sdf.joint_dict:
+                if NAME_SCOPE_SEP is None:
+                    model_name, lname = "", name
+                else:
+                    model_name, jname = name.split(NAME_SCOPE_SEP)
+                sdf_object = C.PDDLSDFObject(
+                    model_name, jname, self.domain.types["sdf::joint"]
+                )
```
Comment (williamshen-nz) [inline lisdf/parsing/pddl_j.py:171]: Should we raise error in an `else` case here? As the lisdf wouldn't satisfy any of the known object types

File: lisdf/parsing/pddl_j.py [hunk 4]
```diff
@@ -144,30 +148,34 @@ def typedvariable(self, name, typename):
...
+                if NAME_SCOPE_SEP is None:
+                    model_name, lname = "", name
+                else:
+                    model_name, lname = name.split(NAME_SCOPE_SEP)
+                sdf_object = C.PDDLSDFObject(
+                    model_name, lname, self.domain.types["sdf::link"]
+                )
+            elif name in self.sdf.joint_dict:
+                if NAME_SCOPE_SEP is None:
+                    model_name, lname = "", name
```
Comment (williamshen-nz) [inline lisdf/parsing/pddl_j.py:166]: I'm guessing `lname` should be `jname` here

File: lisdf/parsing/pddl_j.py [hunk 5]
```diff
@@ -144,30 +148,34 @@ def typedvariable(self, name, typename):
...
+                    name, None, self.domain.types["sdf::model"]
+                )
+            elif name in self.sdf.link_dict:
+                if NAME_SCOPE_SEP is None:
+                    model_name, lname = "", name
+                else:
+                    model_name, lname = name.split(NAME_SCOPE_SEP)
+                sdf_object = C.PDDLSDFObject(
+                    model_name, lname, self.domain.types["sdf::link"]
+                )
```
Comment (williamshen-nz) [inline lisdf/parsing/pddl_j.py:163]: Nit: could probably simplify the elif cases into a single statement. At least the `if NAME_SCOPE else` bit

File: lisdf/parsing/pddl_j.py [hunk 6]
```diff
@@ -144,30 +148,34 @@ def typedvariable(self, name, typename):
...
+                    model_name, jname, self.domain.types["sdf::joint"]
+                )
 
         type = None
         if sdf_object is not None:
             type = sdf_object.sdf_type
         if name in self.problem.objects:
             type = self.problem.objects[name].type
+
         return C.PDDLObject(name, type, sdf_object=sdf_object)
```
Comment (williamshen-nz) [inline lisdf/parsing/pddl_j.py:179]: If `type is None and sdf_object is None` do we want to return an essentially empty object here or raise value error?

File: lisdf/plan_executor/robots/panda.py [hunk 7]
```diff
@@ -18,7 +18,7 @@ def joint_configuration(self) -> np.ndarray:
         return self.configuration[:7]
 
     def set_joint_configuration(self, joint_configuration: np.ndarray) -> None:
-        self.configuration = np.concatenate(
```
Comment (williamshen-nz) [inline lisdf/plan_executor/robots/panda.py:21]: numpy should have type stubs already so mypy shouldn't be complaining here. Are you seeing the errors locally or on github?

Same for other type ignores below

**Review by williamshen-nz — COMMENTED** [review]
Looks great! Only had some minor comments

**Comment by vacancy** [comment]
Thanks for the comments Will.

Can you elaborate more about the "type hint" comment? I can't link that with a specific line...

I have fixed other comments. Please take a look again!

**Review by williamshen-nz — APPROVED** [review]
Thanks for making the changes! Looks good to me :shipit:

**Pull request merged**